### ISSUE TITLE
Potential mismatch on guest OS - hypervisor mappings on upgrades

### ISSUE THREAD

**Issue by nvazquez** [apache/cloudstack#2740]
##### ISSUE TYPE
 * Bug Report

##### COMPONENT NAME
~~~
SCHEMA
~~~

##### CLOUDSTACK VERSION
~~~
4.4.0
4.4.1
4.4.2
4.5.0
4.4.4
4.5.1
4.5.2
4.5.3
4.6.0
4.6.1
4.7.0
4.7.1
4.8.0
4.8.1
4.9.0
4.9.1.0
4.9.2.0
4.9.3.0
4.10.0.0
4.11.0.0
4.11.1.0
4.12.0.0
~~~

##### CONFIGURATION
N/A

##### OS / ENVIRONMENT
N/A

##### SUMMARY
Current support for new guest OS takes place on the upgrade scripts:
- New guest OS are introduced by inserting entries on `guest_os` table with a hardcoded ID.
- Hypervisor support for each guest OS are introduced by inserting entries on `guest_os_hypervisor` table, pointing to the hardcoded guest OS IDs on `guest_os` table.

If an administrator adds at least one new guest OS (using `addGuestOs` API method), mappings after the upgrade will not be consistent (in case new guest OS are supported on the newest CloudStack version), as that API method inserts entries on the `guest_os` table.

##### STEPS TO REPRODUCE
- Add at least one new guest OS on CloudStack
- Upgrade to newest version

##### EXPECTED RESULTS
- Correct mappings for guest OS and hypervisors as stated on the upgrade script

##### ACTUAL RESULTS
Some mapping entries will point to the ones that were created manually and not the ones that were supposed to be added by the upgrade script

**Comment by rhtyd** [comment]
@nvazquez can you advise what is necessary to remedy the issue?

**Comment by DaanHoogland** [comment]
Upgrade scripts should use the new addGuestOs API as well. (my €0,02)

**Comment by rhtyd** [comment]
@nvazquez unless you want to send a PR before 4.13.0.0 freeze, I've moved this to the next major milestone.

**Comment by andrijapanicsb** [comment]
I don't think we'll have time for a proper fix - there is a broader discussion on PR #3609 , lot's of things need to change. I'm closing this one in favour ofthe newer PR #3609
